packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+
+    public get _context() {
+        return this.context;
+    }
+
+    public get containerRuntime() {
+        return this.context.containerRuntime as ContainerRuntime;
+    }
+}
+
+/**
+ * Validates this scenario: When a datastore should be tombstoned that tombstoned and unable to send ops
+ */
+describeNoCompat("GC DataStore Tombstoned When It Is Sweep Ready", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        [],
+    );
+    const inactiveTimeoutMs = 0;
+    const wait = 100;
+    const sweepTimeoutMs = 200;
+
+    const gcOptions: IGCRuntimeOptions = {
+        gcAllowed: true,
+        sweepAllowed: true,
+        snapshotCacheExpiryMs: 0,
+        inactiveTimeoutMs,
+    };
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                maxAckWaitTime: 10000,
+                maxOpsSinceLastSummary: 7000,
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+            },
+        },
+        gcOptions,
+    };
+
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.Test.Tombstone": "true",
+        "Fluid.GarbageCollection.TestOverride.SweepTimeoutMs": sweepTimeoutMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+    const createContainer = async () => provider.createContainer(runtimeFactory, { configProvider });
+    const loadSummarizerAndContainer = async (summaryVersion?: string) => {
+        const absoluteUrl = await mainContainer.getAbsoluteUrl("");
+        return createSummarizerWithContainer(provider, absoluteUrl, runtimeFactory, { configProvider }, summaryVersion);
+    };
+    const summarize = async (summarizer: ISummarizer) => {
+        await provider.ensureSynchronized();
+        return summarizeNow(summarizer);
+    };
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    const createUnreferencedDataStore = async () => {
```

Comment:
The name and what happens in the function is very different. Try to name it such that it reflects the data store is unreferenced in summary and the summary id is also returned. Also, add a comment explaining what this function does as its not straightforward.